server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -701,19 +701,40 @@ public String[] indexAliases(
         IndexAbstraction ia = state.metadata().getIndicesLookup().get(index);
         DataStream dataStream = ia.getParentDataStream();
         if (dataStream != null) {
+            if (skipIdentity == false && resolvedExpressions.contains(dataStream.getName())) {
+                return null;
+            }
```

Comment:
This enables us to skip the filters when the request targets the data stream name and a filtered alias

Reply:
Can you add this as a comment in the code?